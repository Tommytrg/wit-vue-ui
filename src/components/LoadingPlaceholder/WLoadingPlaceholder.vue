<template>
  <div
    class="placeholder bg-gray-50 bg-opacity-60 bg-gradient-to-r from-gray-50 shadow-sm shadow-gray-100 bg-[length:200%_200%]"
  >
    &nbsp;
  </div>
  <span class="sr-only">Loading...</span>
</template>
<script setup lang="ts">
defineProps({
  height: {
    type: String,
    required: true
  },
  width: {
    type: String,
    required: true
  },
  borderRadius: {
    type: String,
    required: true
  }
})
</script>
<style scoped lang="scss">
@keyframes bgAnimate {
  0% {
    background-position: 50% 0;
  }
  100% {
    background-position: -150% 0;
  }
}
.placeholder {
  height: v-bind(height);
  width: v-bind(width);
  border-radius: v-bind(borderRadius);
  animation: bgAnimate 1.2s linear infinite;
}
</style>
