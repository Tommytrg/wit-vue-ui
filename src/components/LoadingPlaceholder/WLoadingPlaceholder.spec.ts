import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import WLoadingPlaceholder from './WLoadingPlaceholder.vue'

describe('WLoadingPlaceholder', () => {
  it('renders properly', () => {
    const wrapper = mount(WLoadingPlaceholder, {
      props: {
        height: '100px',
        width: '150px',
        borderRadius: '8%'
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
