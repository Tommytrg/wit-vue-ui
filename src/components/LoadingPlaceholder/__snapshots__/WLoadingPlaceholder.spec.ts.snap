// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WLoadingPlaceholder > renders properly 1`] = `
<div
  data-v-app=""
>
  
  <div
    class="placeholder bg-gray-50 bg-opacity-60 bg-gradient-to-r from-gray-50 shadow-sm shadow-gray-100 bg-[length:200%_200%]"
    data-v-bd9b9d04=""
  >
    Â 
  </div>
  <span
    class="sr-only"
    data-v-bd9b9d04=""
  >
    Loading...
  </span>
  
</div>
`;
